From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aswin-mcw <aswin.kalies@multicorewareinc.com>
Date: Wed, 29 Jul 2026 15:27:40 +0530
Subject: [PATCH 092/109] fixup! - Move root to /usr. - Change sorting mount
 points. - By default mount without ACLs. - Can read /etc/fstab with short
 mount point format.

Cygwin: testsuite: install the test runtime under testinst/usr/bin

Mirror the real MSYS2 layout so init_installation_root() resolves to testinst.
MSYS2 strips three levels from the DLL path (<root>/usr/bin/msys-2.0.dll -> <root>);
with the DLL and helpers under testinst/bin the extra strip overshot the root to
${builddir}, so /tmp resolved to ${builddir}/tmp instead of testinst/tmp.  Put the
test DLL and the sh/sleep/ls helpers under testinst/usr/bin and point runtime_root
there; / -> testinst, /bin and /usr/bin -> testinst/usr/bin, /tmp stays testinst/tmp.

(cherry picked from commit 60cbfd3c20eca9cfe0dfca32697f947bd054ab2c)
(cherry picked from commit dc07145855015fb460eb7805b12ffd8b78e65b67)
---
 winsup/testsuite/Makefile.am | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/winsup/testsuite/Makefile.am b/winsup/testsuite/Makefile.am
index 7e0ea2fe0..5f2cc1b4b 100644
--- a/winsup/testsuite/Makefile.am
+++ b/winsup/testsuite/Makefile.am
@@ -372,9 +372,9 @@ export cygrun=$(builddir)/mingw/cygrun
 
 check-local:
 	$(MKDIR_P) ${builddir}/testinst/tmp
-	cd ${builddir}/testinst/usr/bin && cp /usr/libexec/busybox/bin/busybox.exe sh.exe
-	cd ${builddir}/testinst/usr/bin && cp /usr/libexec/busybox/bin/busybox.exe sleep.exe
-	cd ${builddir}/testinst/usr/bin && cp /usr/libexec/busybox/bin/busybox.exe ls.exe
+	cd ${builddir}/testinst/usr/bin && cp /usr/bin/busybox.exe sh.exe
+	cd ${builddir}/testinst/usr/bin && cp /usr/bin/busybox.exe sleep.exe
+	cd ${builddir}/testinst/usr/bin && cp /usr/bin/busybox.exe ls.exe
 
 # target to build all the programs needed by check, without running check
 check_programs: $(check_PROGRAMS)
